### DISCUSSION TITLE
SQS Inconsistency

### DISCUSSION THREAD

**Discussion by whatadewitt** [serverless/serverless#11170] [General]
I'm having some weird inconsistencies with my functions that are being triggered by SQS queues where messages aren't being deleted after being processed. What is it exactly that will ensure that serverless will remove the message from the queue after processing? Should I be implementing `sqs.deleteMessage` manually in my function?

Thanks!

**Comment by TroyDavidson** [comment]
Yes, you need to mark the messages for deletion if you've handled them correctly.  

If you don't do anything, the timeout will happen and the messages will become visible again in the queue.

**Comment by pgrzesik** [comment]
The situation that @TroyDavidson mentions would be the case if messages were manually pulled from SQS, but if you're using native Lambda <-> SQS integration, then the messages will be automatically deleted if your Lambda function successfully returns.

Maybe you're hitting some timeouts of your Lambda function or it's running into some errors? In such case, the messages won't be deleted and will be processed again.
